presto-parser/src/main/java/com/facebook/presto/sql/parser/AstBuilder.java

```diff
@@ -663,7 +664,67 @@ public Node visitShowColumns(SqlBaseParser.ShowColumnsContext context)
+        // The following properties of SELECT subquery are required:
+        //  - only one relation in FROM
+        //  - only plain columns in projection
+        //  - only plain columns and constants in WHERE
+        //  - no group by
+        //  - no having
+        //  - no set quantifier
+        List<SqlBaseParser.RelationContext> relations = context.querySpecification().relation();
+        check(relations.size() != 0, "There must be exactly one relation in SHOW STATS SELECT clause", context.querySpecification());
+        check(relations.size() == 1, "There must be exactly one relation in SHOW STATS SELECT clause", context.querySpecification().relation(1));
+        check(context.querySpecification().groupBy() == null, "GROUP BY is not supported in SHOW STATS SELECT clause", context.querySpecification().groupBy());
+        check(context.querySpecification().having == null, "HAVING is not supported in SHOW STATS SELECT clause", context.querySpecification().having);
+        check(context.querySpecification().setQuantifier() == null, "ALL/DISTINCT are not supported by SHOW STATS SELECT clause", context.querySpecification().setQuantifier());
+
+        QuerySpecification specification = (QuerySpecification) visitQuerySpecification(context.querySpecification());
+        int selectId = 0;
+        for (SelectItem selectItem : specification.getSelect().getSelectItems()) {
+            if (selectItem instanceof AllColumns) {
+                continue;
+            }
+            check(selectItem instanceof SingleColumn, "Only SingleColumn and AllColumn selects are supported by SHOW STATS SELECT clause", context.querySpecification().selectItem(selectId));
+
+            SingleColumn columnSelect = (SingleColumn) selectItem;
+            check(columnSelect.getExpression() instanceof QualifiedNameReference, "Only FieldReference is allowed on SELECT list of SHOW STATS SELECT clause", context.querySpecification().selectItem(selectId));
+
+            selectId++;
+        }
+
+        if (specification.getWhere().isPresent()) {
+            validateShowStatsWhereExpression(specification.getWhere().get(), context.querySpecification().where);
+        }
+
+        Query query = new Query(Optional.empty(), specification, ImmutableList.of(), Optional.empty());
+
+        return new ShowStats(Optional.of(getLocation(context)), query);
+    }
+
+    public static final List<Class<? extends Expression>> ALLOWED_SHOW_STATS_WHERE_EXPRESSION_TYPES = ImmutableList.of(
+            Literal.class, QualifiedNameReference.class, ComparisonExpression.class, LogicalBinaryExpression.class, NotExpression.class);
+
+    void validateShowStatsWhereExpression(Expression expression, ParserRuleContext context)
+    {
+        check(ALLOWED_SHOW_STATS_WHERE_EXPRESSION_TYPES.stream().anyMatch(clazz -> clazz.isInstance(expression)), "Only Literals, FieldReferences, Comparisons, And, Ors and Not expression are allowed in WHERE of SHOW STATS SELECT clause", context);
```

Comment:
`Only literals, column references, comparators and logical operators are allowed ...`